.github/workflows/all.yml

```diff
@@ -296,35 +302,47 @@ jobs:
           rmdir .\packaging\k6-$env:VERSION-windows-amd64\
       - name: Add signtool to PATH
         run: echo "${env:ProgramFiles(x86)}\Windows Kits\10\bin\x64" | Out-File -FilePath $env:GITHUB_PATH -Append
-      - name: Convert base64 certificate to PFX
-        run: |
-          $bytes = [Convert]::FromBase64String("${{ secrets.WIN_SIGN_CERT }}")
-          [IO.File]::WriteAllBytes("k6.pfx", $bytes)
-      - name: Sign Windows binary
-        run: signtool sign /f k6.pfx /p "${{ secrets.WIN_SIGN_PASS }}" /tr "http://timestamp.digicert.com" /td sha256 /fd sha256 "packaging\k6.exe"
-      - name: Create MSI package
+
+      - name: Create the MSI package
         run: |
           $env:VERSION = $env:VERSION -replace 'v(\d+\.\d+\.\d+).*','$1'
           pandoc -s -f markdown -t rtf -o packaging\LICENSE.rtf LICENSE.md
           cd .\packaging
           candle.exe -arch x64 "-dVERSION=$env:VERSION" k6.wxs
           light.exe -ext WixUIExtension k6.wixobj
-      - name: Sign MSI package
-        run: signtool sign /f k6.pfx /p "${{ secrets.WIN_SIGN_PASS }}" /tr "http://timestamp.digicert.com" /td sha256 /fd sha256 "packaging\k6.msi"
-      - name: Cleanup signing artifacts
-        run: del k6.pfx
+
+      - name: Sign Windows binary and .msi package
+        # Only sign packages that are built from master or a version tag
+        if: ${{ github.ref == 'refs/heads/master' || startsWith(github.ref, 'refs/tags/v') }}
+        run: |
+          # Convert base64 certificate to PFX
+          $bytes = [Convert]::FromBase64String("${{ secrets.WIN_SIGN_CERT }}")
+          [IO.File]::WriteAllBytes("k6.pfx", $bytes)
+
+          # Sign the Windows binary
+          signtool sign /f k6.pfx /p "${{ secrets.WIN_SIGN_PASS }}" /tr "http://timestamp.digicert.com" /td sha256 /fd sha256 "packaging\k6.exe"
+
+          # Sign the MSI package
+          signtool sign /f k6.pfx /p "${{ secrets.WIN_SIGN_PASS }}" /tr "http://timestamp.digicert.com" /td sha256 /fd sha256 "packaging\k6.msi"
+
+          # Cleanup signing artifacts
+          del k6.pfx
+
       - name: Prepare Chocolatey package
         run: |
-          $env:VERSION = $env:VERSION.TrimStart("v", " ")
+          $env:VERSION = $env:VERSION -replace 'v(\d+\.\d+\.\d+).*','$1'
           cd .\packaging
           (Get-Content '.\k6.portable.nuspec' -Raw).Replace("<version>__REPLACE__</version>", "<version>$env:VERSION</version>") | Out-File '.\k6.portable.nuspec'
+
```

Comment:
@na-- as an alternative, if we prefer to keep it, I think we can just disable the stage commenting the code.

Reply:
@oleiade, a k6 package already exists in winget (see https://github.com/microsoft/winget-pkgs/tree/master/manifests/k/k6/k6 and https://winstall.app/apps/k6.k6) and chocolately (see https://community.chocolatey.org/packages/k6)

However, we are not the ones maintaining these packages, they are updated by external packagers and contributors, similar to the homebrew package (see https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/k6.rb and https://formulae.brew.sh/formula/k6) and the AUR k6 package (https://aur.archlinux.org/packages/k6). We have absolutely no control over them and they don't use our CI processes.

The Choco (or, more accurately, NuGet) package in our CI pipeline is a remnant from a long time ago when we used to publish it on BinTray. We [haven't done this since k6 v0.32.0](https://github.com/grafana/k6/blob/master/release%20notes/v0.32.0.md#move-out-of-bintray), since they JFrog discontinued BinTray, but we [used to have a semi-official Choco package](https://github.com/grafana/k6/tree/v0.31.1#windows) with some very complicated instructions that don't seem to have survived anywhere ([this](https://bintray.com/repo/buildSettings?repoPath=%2Floadimpact%2Fchoco) was the old link). So this CI package is currently literally getting [built](https://github.com/grafana/k6/blob/49cdd8e4282aa7e1da8d9c12982e03af553072e1/.github/workflows/all.yml#L331-L340) and then [deleted](https://github.com/grafana/k6/blob/49cdd8e4282aa7e1da8d9c12982e03af553072e1/.github/workflows/all.yml#L393-L394) before it gets published... 

So, having explained all of this, I think my personal vote has been switched to :-1:, that is, to remove it... :sweat_smile: @codebien, I'd prefer not to comment it out, but maybe remove it in a dedicated commit so it's easy to recover in the future if we need to. Commenting it out will just get in the way without adding any value.

Reply:
I am also for just dropping it (in a separate commit if possible)

Reply:
Fair enough @na--  thanks for the explanation ! Then I'm for dropping it too 🤝

Reply:
Resolved by https://github.com/grafana/k6/pull/2591/commits/1541175e67573dc01a42051c16c48c6465086093